var/spack/repos/builtin/packages/vtk-m/package.py

```diff
@@ -220,6 +222,19 @@ def cmake_args(self):
             if "+rocm" in spec:
                 options.append(self.builder.define_hip_architectures(self))
 
+            # Add support for OpenMP with external clang
+            if spec.satisfies("%clang +openmp"):
```

Comment:
It seems not all CMakePackages want to use the same CMake OpenMP variables:

Most do this to use a libomp built using brew on MacOS, while others use it set the args
to use the libomp from `llvm-openmp` like [var/spack/repos/builtin/packages/onednn/package.py#L133](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/onednn/package.py#L133)

And, instead of `cmakepackage.py`, which would not want to have package-specfic setup, something like `setup_dependent_package` could be used in `llvm-openmp` to set the `module.llvm_openmp_cmake_args` for dependents and then use `inspect.getmodule` to get the dependents to get the common OpenMP CMmake args for it. @alalazo?

To stop the GitLab pipeline for now, I'm setting this to draft until there is agreement.

@vicentebolea In case this stalls this PR too long, feel free to force-push your initial change and follow-up with a new PR.

Reply:
@bernhardkaindl thanks, it is taking long, I repushed the original changes. Please followup with your changes in a new PR.